### DISCUSSION TITLE
6. 웹 관련 네트워크

### DISCUSSION THREAD

**Discussion by seongeun42** [Shortudy-10th/tech-for-developer#160] [질의 응답]
http, 브라우저 동작 등

**Comment by LireEruel** [comment]
## Q. 주소창에 사용자에 naver.com을 입력하면, 화면에 보여지기까지 브라우저가 렌더링하는 과정을 얘기해보세요.

1. 웹 서버에 리소스 요청 및 수신
url을 입력하면, 브라우저는 해당 웹 서버에 요청을 보냅니다. 웹 서버는 요청을 받고 해당 페이지의 HTML 문서를 응답으로 반환합니다.
2. HTML 파싱
HTML 문서를 파싱하여 DOM Tree를 구축합니다.
3. 외부 리소스 (CSS, JS, assets) 요청 및 수신
4. CSS 파싱
외부 CSS파일과 내장된 스타일을 파싱하여 CSSOM (CSS Object Model)트리를 구축합니다.
5. 렌더 트리 생성
DOM 트리와 CSSOM 트리를 결합하여 렌더트리를 생성합니다.
6. JavaScript 실행
7. 레이아웃
렌더트리를 기반으로 각 노드의 위치나 크기를 계산하여 레이아웃을 결정합니다.
8. 페인트
렌더트리의 각 노드를 화면에 그립니다.

**Comment by LireEruel** [comment]
## HTTP와 HTTPS의 차이점은 무엇인가요?

HTTP는 Hypertext Transfer Protocol 이고 HTTPS는 HTTP에 Secure이 붙은 것 입니다.
 HTTPS는 웹 사이트가 신뢰할 수 있는 인증 기관(Certificate Authority)에 의해 발급된 SSL/TLS 인증서를 사용합니다.

**Comment by LireEruel** [comment]
## Q. 웹 서버와 웹 어플리케이션 서버의 차이점

웹 서버는 주로 정적인 콘텐츠를 제공하는 역할을 합니다. 이 서버는 클라이언트로부터 HTTP 요청을 받아들이고, 주로 HTML, CSS, 이미지 등과 같은 정적 파일을 응답으로 제공합니다. 웹 서버는 요청된 파일을 파일 시스템에서 찾아내어 클라이언트에게 전달합니다.

웹 애플리케이션 서버는 동적인 콘텐츠를 생성하고 제공하는 역할을 합니다. 주로 웹 애플리케이션의 핵심 로직을 실행하고 그 결과를 클라이언트에게 돌려주는 역할을 합니다.
